### ISSUE TITLE
please issue on make (High Sierra MacPro NVIDIA 1080Ti 11GB)

### ISSUE THREAD

**Issue by ludogemma** [ryanfb/homebrew-alicevision#9]
Hi there i can't make the alicevision..

MacPro High Sierra 10.13.6
NVIDIA 1080Ti 11GB
Cuda Toolkit installed 10.1
Xcode commandlinetools for Xcode 10.1

tried the formula: brew install ryanfb/alicevision/alicevision

but i receive this error at start of the make:

==> cmake .. -DBoost_NO_BOOST_CMAKE=ON -DALICEVISION_USE_OPENMP:BOOL=OFF -DALICEVISION_USE_ALEMBIC:BOOL=ON -DALICEVISION_BUILD_DOC:BOOL=OFF -DFLANN_INCLUDE_DI
==> make install
Last 15 lines from /Users/sbarrile/Library/Logs/Homebrew/alicevision/02.make:
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [Darwin-x86_64/MeshDenoiser] Error 1
make[1]: *** [src/dependencies/MeshSDFilter/CMakeFiles/MeshDenoiser.dir/all] Error 2

**Comment by ryanfb** [comment]
I need the full logs from `/Users/sbarrile/Library/Logs/Homebrew/alicevision/02.make` to be able to diagnose anything.

**Comment by ludogemma** [comment]
sorry here we go:

(base) ->> Sbarrile $ brew install alicevision
Updating Homebrew...
==> Auto-updated Homebrew!
Updated 1 tap (homebrew/core).
No changes to formulae.

==> Installing alicevision from ryanfb/alicevision
==> Downloading https://github.com/alicevision/osi_clp/archive/38ab28d1c5a5
Already downloaded: /Users/sbarrile/Library/Caches/Homebrew/downloads/868af2070136f5cd5bccef47f0d69bcdd0c1bbecc3ef5387aa2f5cd624f48679--osi_clp-38ab28d1c5a53de13c8684cdc272b1deb8cef459.tar.gz
==> Downloading https://github.com/alicevision/MeshSDFilter/archive/b7dfeed
Already downloaded: /Users/sbarrile/Library/Caches/Homebrew/downloads/8e567e6974bcf2d357553e8157c07c6e022f3833c46cd0a1aea6bd8bffeaa7d9--MeshSDFilter-b7dfeed64be90f2eff49345cf65451b700d3a417.tar.gz
==> Downloading https://github.com/alicevision/nanoflann/archive/cc77e17934
Already downloaded: /Users/sbarrile/Library/Caches/Homebrew/downloads/ac4a14dcf56699c2ecd92da129f982db91ff1fa8abe9bcd5b486328431ed9528--nanoflann-cc77e17934441dc82b33fd00e0a8a1398f24c928.tar.gz
==> Downloading https://github.com/alicevision/AliceVision/archive/v2.2.0.t
Already downloaded: /Users/sbarrile/Library/Caches/Homebrew/downloads/18c5e714b3e20de37f6c56a821e2dd5a994c372f74c63a76ebe64b90b9b997d0--AliceVision-2.2.0.tar.gz
==> Patching
patching file src/CMakeLists.txt
Hunk #1 succeeded at 195 (offset 14 lines).
patching file src/software/convert/main_convertLDRToHDR.cpp
==> cmake .. -DBoost_NO_BOOST_CMAKE=ON -DALICEVISION_USE_OPENMP:BOOL=OFF -D
==> make install
Last 15 lines from /Users/sbarrile/Library/Logs/Homebrew/alicevision/02.make:
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [Darwin-x86_64/MeshDenoiser] Error 1
make[1]: *** [src/dependencies/MeshSDFilter/CMakeFiles/MeshDenoiser.dir/all] Error 2
[ 59%] Linking CXX executable ../../../Darwin-x86_64/aliceVision_convertLDRToHDR
cd /tmp/alicevision-20200721-80923-1tr6aq3/AliceVision-2.2.0/build/src/software/convert && /usr/local/Cellar/cmake/3.17.3/bin/cmake -E cmake_link_script CMakeFiles/aliceVision_convertLDRToHDR.dir/link.txt --verbose=1
/usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang++   -march=westmere -msse2 -msse3 -mssse3 -msse4.1 -msse4.2 -mno-sse4a -mno-avx -mno-fma -mno-bmi2 -mno-avx2 -mno-xop -mno-fma4 -mno-avx512f -mno-avx512vl -mno-avx512pf -mno-avx512er -mno-avx512cd -mno-avx512dq -mno-avx512bw -mno-avx512ifma -mno-avx512vbmi -O3 -Wl,-search_paths_first -Wl,-headerpad_max_install_names  CMakeFiles/aliceVision_convertLDRToHDR.dir/main_convertLDRToHDR.cpp.o  -o ../../../Darwin-x86_64/aliceVision_convertLDRToHDR-0.1  -Wl,-rpath,/tmp/alicevision-20200721-80923-1tr6aq3/AliceVision-2.2.0/build/Darwin-x86_64 ../../../Darwin-x86_64/libaliceVision_hdr.2.2.dylib /usr/local/lib/libboost_atomic-mt.dylib /usr/local/lib/libboost_container-mt.dylib /usr/local/lib/libboost_date_time-mt.dylib /usr/local/lib/libboost_filesystem-mt.dylib /usr/local/lib/libboost_graph-mt.dylib /usr/local/lib/libboost_log-mt.dylib /usr/local/lib/libboost_log_setup-mt.dylib /usr/local/lib/libboost_program_options-mt.dylib /usr/local/lib/libboost_regex-mt.dylib /usr/local/lib/libboost_serialization-mt.dylib /usr/local/lib/libboost_system-mt.dylib /usr/local/lib/libboost_thread-mt.dylib /usr/local/lib/libboost_timer-mt.dylib /usr/local/lib/libboost_chrono-mt.dylib ../../../Darwin-x86_64/libaliceVision_image.2.2.dylib ../../../Darwin-x86_64/libaliceVision_system.2.2.dylib ../../../Darwin-x86_64/libaliceVision_numeric.2.2.dylib /usr/local/lib/libOpenImageIO.dylib /usr/local/lib/libImath-2_5.dylib /usr/local/lib/libIex-2_5.dylib /usr/local/lib/libHalf-2_5.dylib /usr/local/lib/libIlmThread-2_5.dylib -ldl /usr/local/lib/libboost_date_time-mt.dylib /usr/local/lib/libboost_log-mt.dylib /usr/local/lib/libboost_log_setup-mt.dylib /usr/local/lib/libboost_program_options-mt.dylib /usr/local/lib/libboost_system-mt.dylib /usr/local/lib/libboost_thread-mt.dylib 
cd /tmp/alicevision-20200721-80923-1tr6aq3/AliceVision-2.2.0/build/src/software/convert && /usr/local/Cellar/cmake/3.17.3/bin/cmake -E cmake_symlink_executable ../../../Darwin-x86_64/aliceVision_convertLDRToHDR-0.1 ../../../Darwin-x86_64/aliceVision_convertLDRToHDR
[ 59%] Built target aliceVision_convertLDRToHDR
[ 59%] Linking CXX shared library ../../../Darwin-x86_64/libaliceVision_matching.dylib
cd /tmp/alicevision-20200721-80923-1tr6aq3/AliceVision-2.2.0/build/src/aliceVision/matching && /usr/local/Cellar/cmake/3.17.3/bin/cmake -E cmake_link_script CMakeFiles/aliceVision_matching.dir/link.txt --verbose=1
/usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang++  -march=westmere -msse2 -msse3 -mssse3 -msse4.1 -msse4.2 -mno-sse4a -mno-avx -mno-fma -mno-bmi2 -mno-avx2 -mno-xop -mno-fma4 -mno-avx512f -mno-avx512vl -mno-avx512pf -mno-avx512er -mno-avx512cd -mno-avx512dq -mno-avx512bw -mno-avx512ifma -mno-avx512vbmi -O3 -dynamiclib -Wl,-headerpad_max_install_names -compatibility_version 2.0.0 -current_version 2.2.0 -o ../../../Darwin-x86_64/libaliceVision_matching.2.2.dylib -install_name @rpath/libaliceVision_matching.2.dylib CMakeFiles/aliceVision_matching.dir/io.cpp.o CMakeFiles/aliceVision_matching.dir/matcherType.cpp.o CMakeFiles/aliceVision_matching.dir/RegionsMatcher.cpp.o  -Wl,-rpath,/tmp/alicevision-20200721-80923-1tr6aq3/AliceVision-2.2.0/build/Darwin-x86_64 ../../../Darwin-x86_64/libaliceVision_feature.2.2.dylib /usr/local/lib/libboost_filesystem-mt.dylib /usr/local/lib/libflann_cpp.dylib ../../../Darwin-x86_64/libaliceVision_image.2.2.dylib ../../../Darwin-x86_64/libaliceVision_numeric.2.2.dylib /usr/local/lib/libOpenImageIO.dylib /usr/local/lib/libImath-2_5.dylib /usr/local/lib/libIex-2_5.dylib /usr/local/lib/libHalf-2_5.dylib /usr/local/lib/libIlmThread-2_5.dylib -ldl ../../../Darwin-x86_64/libaliceVision_gpu.2.2.dylib ../../../Darwin-x86_64/libaliceVision_system.2.2.dylib /usr/local/lib/libboost_log-mt.dylib /usr/local/lib/libboost_thread-mt.dylib /usr/local/lib/libboost_log_setup-mt.dylib /usr/local/lib/libboost_system-mt.dylib /usr/local/lib/libboost_date_time-mt.dylib /usr/local/lib/libboost_program_options-mt.dylib ../../../Darwin-x86_64/libvlsift.a /usr/local/lib/libboost_container-mt.dylib 
cd /tmp/alicevision-20200721-80923-1tr6aq3/AliceVision-2.2.0/build/src/aliceVision/matching && /usr/local/Cellar/cmake/3.17.3/bin/cmake -E cmake_symlink_library ../../../Darwin-x86_64/libaliceVision_matching.2.2.dylib ../../../Darwin-x86_64/libaliceVision_matching.2.dylib ../../../Darwin-x86_64/libaliceVision_matching.dylib
[ 59%] Built target aliceVision_matching
make: *** [all] Error 2

If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
  https://github.com/ryanfb/homebrew-alicevision/issues

**Comment by ludogemma** [comment]
[02.make.zip](https://github.com/ryanfb/homebrew-alicevision/files/4953613/02.make.zip)

**Comment by ludogemma** [comment]
sorry i didn't understand about the log .. but i attached zipped the file, thanks!

**Comment by ludogemma** [comment]
> I need the full logs from `/Users/sbarrile/Library/Logs/Homebrew/alicevision/02.make` to be able to diagnose anything.

execuse me was ok the file 02.make.zip or u need in other way?

**Comment by ryanfb** [comment]
I've started looking into it, and after fixing the first issue (OpenMP is now detected but not linked in the MeshSDFilter dependency) there seem to be other issues afterwards, perhaps related to upgrading AliceVision to 2.2.0. If I can't solve them I'll just roll back to a previous working version.

**Comment by ludogemma** [comment]
> I've started looking into it, and after fixing the first issue (OpenMP is now detected but not linked in the MeshSDFilter dependency) there seem to be other issues afterwards, perhaps related to upgrading AliceVision to 2.2.0. If I can't solve them I'll just roll back to a previous working version.

**Comment by ludogemma** [comment]
ok thanks, so i wait your fix :)
at least is not a problem related to my system.

**Comment by ludogemma** [comment]
sorry about close and reopen .. i'm a bit noob on github..

**Comment by ludogemma** [comment]
ryan i was able to install alicevision reading this comments:
https://github.com/ryanfb/homebrew-alicevision/issues/5#issuecomment-544188194

and i just changed the versions with the ones downloaded:

`cmake \
-DAlembic_DIR:PATH=/path/to/alembic/build/lib/Alembic \
-DCeres_DIR:PATH=/usr/local/Cellar/ceres-solver/1.14.0_12/lib/cmake/Ceres/ \
-DFLANN_INCLUDE_DIR_HINTS:PATH=/usr/local/Cellar/flann/1.9.1_9/include/ \
-DCMAKE_MODULE_PATH:PATH=/usr/local/Cellar/eigen/3.3.7/share/cmake/Modules/ \
-DOPENEXR_HOME:PATH=/usr/local/Cellar/openexr/2.5.2/ \
-DOPENIMAGEIO_LIBRARY_DIR_HINTS:PATH=/usr/local/Cellar/openimageio/2.1.17/lib/ \
-DOPENIMAGEIO_INCLUDE_DIR:PATH=/usr/local/Cellar/openimageio/2.1.17/include/ \
-DCMAKE_BUILD_TYPE=Release \
-DCMAKE_C_COMPILER=gcc \-DCMAKE_CXX_COMPILER=g++ \
-DOpenMP_CXX_FLAGS="-Xpreprocessor -fopenmp -lomp -L'$(brew --prefix libomp)/lib' -I'$(brew --prefix libomp)/include'" \
-DOpenMP_CXX_LIB_NAMES="omp" \
-DOpenMP_omp_LIBRARY=$(brew --prefix libomp)/lib/libomp.a \
..
` 

and worked... so maybe this message could help you about the formula :)

**Comment by ludogemma** [comment]
@ryanfb excuse me in your blog you also wrote:

"Third, CUDA 10.2 is only compatible with the version of clang distributed with Xcode 10.2"
but i downloaded the XCODE 10.2 on my HIgh Sierra and it told me isn't compatible with my system and requires i guess 10.14 (i guess mojave), so how you could install xcode on high sierra?
thanks in advance

**Comment by ryanfb** [comment]
I updated the guide for CUDA 10.2 based on the [CUDA 10.2 docs](https://docs.nvidia.com/cuda/archive/10.2/cuda-installation-guide-mac-os-x/index.html) which say it's only compatible with Xcode 10.2 on High Sierra. Looks like I'm still running CUDA 10.1 with Xcode 10.1, so I haven't tried it myself.

**Comment by urbanstarship** [comment]
@ryanfb @ludogemma if either of you can assist, that would be great.  I'm stuck on the same error.  I don't know enough about Unix to follow the advice about CMake and Alembic.

**Comment by ryanfb** [comment]
I just pushed a commit that fixed the build on my machine. Ultimately, I had to disable MeshSDFilter. The only impact should be that MeshDenoising and MeshDecimate aren't built.

If someone can fix AliceVision building on OS X with MeshSDFilter enabled, feel free to make a PR for it. I left my work-in-progress (add MeshSDFilter to the include path and use a forced-non-OpenMP build of MeshSDFilter)—but CMake needs to also know where to link MeshSDFilter when it's linking together MeshDenoising/MeshDecimate.
